indexing-hadoop/src/main/java/org/apache/druid/indexer/HadoopIngestionSpec.java

```diff
@@ -69,6 +73,7 @@ public HadoopIngestionSpec(
     this.ioConfig = ioConfig;
     this.tuningConfig = tuningConfig == null ? HadoopTuningConfig.makeDefaultTuningConfig() : tuningConfig;
     this.uniqueId = uniqueId == null ? UUIDUtils.generateUuid() : uniqueId;
+    this.context = context == null ? new HashMap<>() : new HashMap<>(context);
```

Comment:
If context is not passed in, should it be left as `null` or `ImmutableMap.of()`?

Reply:
If the context is not set by the user then we set it to mutable empty hashmap. This allow downstream (internal Druid code) to set the context if needed. For example, in batch ingestion (non-hadoop) even if context is not set by the user, we set it to a mutable empty hashmap. This allow downstream processing to determine contexts like forceTimeChunkLock and set it appropriately to the context map.

Reply:
@maytasm and I discussed offline and we agreed that to match with other batch ingestion methodologies,  context could be set as an empty mutable map.